internal/broadcast/message.go

```diff
@@ -19,31 +19,40 @@ package broadcast
 import (
 	"context"
 
-	"github.com/hyperledger-labs/firefly/internal/config"
 	"github.com/hyperledger-labs/firefly/internal/i18n"
 	"github.com/hyperledger-labs/firefly/internal/log"
 	"github.com/hyperledger-labs/firefly/pkg/database"
 	"github.com/hyperledger-labs/firefly/pkg/fftypes"
 )
 
 func (bm *broadcastManager) BroadcastMessage(ctx context.Context, ns string, in *fftypes.MessageInOut, waitConfirm bool) (out *fftypes.Message, err error) {
-	return bm.BroadcastMessageWithID(ctx, ns, nil, in, nil, waitConfirm)
+	return bm.BroadcastMessageWithID(ctx, ns, nil, in, nil, waitConfirm, false)
 }
 
-func (bm *broadcastManager) BroadcastMessageWithID(ctx context.Context, ns string, id *fftypes.UUID, unresolved *fftypes.MessageInOut, resolved *fftypes.Message, waitConfirm bool) (out *fftypes.Message, err error) {
+func (bm *broadcastManager) BroadcastMessageWithResolvedID(ctx context.Context, ns string, in *fftypes.MessageInOut, waitConfirm bool) (out *fftypes.Message, err error) {
+	return bm.BroadcastMessageWithID(ctx, ns, nil, in, nil, waitConfirm, true)
+}
+
+// THIS IS WHERE THE NEW CODE FIRST CALLS ResolveInputIdentity
+// IT already has an author and key here
+func (bm *broadcastManager) BroadcastMessageWithID(ctx context.Context, ns string, id *fftypes.UUID, unresolved *fftypes.MessageInOut, resolved *fftypes.Message, waitConfirm bool, identityResolved bool) (out *fftypes.Message, err error) {
 	if unresolved != nil {
 		resolved = &unresolved.Message
 	}
 	resolved.Header.ID = id
 	resolved.Header.Namespace = ns
 	resolved.Header.Type = fftypes.MessageTypeBroadcast
-	if resolved.Header.Author == "" {
-		resolved.Header.Author = config.GetString(config.OrgIdentity)
-	}
 	if resolved.Header.TxType == "" {
 		resolved.Header.TxType = fftypes.TransactionTypeBatchPin
 	}
 
+	if !identityResolved {
```

Comment:
Personally I don't like when bits of a method are enabled/disabled with a boolean flag - feels like it should just be a separate method that you invoke beforehand if you need it.

Side note, after this block executes, the identity _is_ resolved, so it's odd that the boolean is propagated on line 75 rather than being flipped to `true`.

Reply:
Think this is because the `BroadcastRootOrgDefinition` method was separated out, which is missing coverage.
Addressing this.